lib/event-listeners/food-listener.js

```diff
@@ -6,18 +6,38 @@ import { deleteFood } from '../ajax-requests/delete-food.js'
 
 const $ = require('jquery')
 
-let foodObj = $("#food_form").on("submit", function(event) {
+let name = $("#name").val()
+let calories =  $("#calories").val()
+
+$("#food_form").on("submit", function(event) {
   event.preventDefault()
-  let name = $("#name").val()
-  let calories =  $("#calories").val()
-  let newFood = new Food(name, calories)
-  $("#name").val("")
-  $("#calories").val("")
-  createFood(newFood).done(function( data ) {
-    appendFood(data)
-  })
+  let newFood = foodFormData()
+  clearFormFields()
+  if (foodFormHasData()) {
```

Comment:
I make this check before creating the new food and clearing the form fields. would probably wrap everything in this with that check